app/controllers/products_controller.rb

```diff
@@ -0,0 +1,111 @@
+class ProductsController < ApplicationController
+  before_action :current_user
+  before_action :require_login, only: [:new, :create, :edit, :update, :retire]
+  before_action :check_user_product, only: [:review]
+  before_action :check_user_id, only: [:retire, :edit, :update]
+
+  def index
+    @categories = Category.all
+    @merchants = User.all
+    @products = Product.order(:created_at).where(retired: false)
+    case params[:type]
+
+      when "merch"
+        @products = @products.where("user_id = #{params[:order]}")
+      when "cat"
+        prod = Product.where(retired: false)
+        @products = prod.select {|product|
+          rows = product.categories.where("category_id = #{params[:order]}")
+          !rows.to_a.empty?
+        }
+      else
+        #binding.pry
+        @order = "prod"
+    end
+  end
+
+  def show
+    @product = Product.find(params[:id])
+    @review = Review.new
+    @reviews = Review.where("product_id = #{params[:id]}").reverse
+    @order_item = my_order.order_items.new
+  end
+
+  def review
+    review = Review.create(review_params)
+    flash[:rating] = "You must give your review a star rating."
+    redirect_to product_path(params[:product_id], anchor: "review")
+  end
+
+  def new
+    @product = Product.new
+    @action = "create"
+    @categories = @product.categories
+    @all_categories = Category.all
+  end
+
+  def create
+    @user_id = session[:user_id]
+    @product = Product.new(product_params)
+    if !params[:categories].nil?
```

Comment:
It looks like you want the if statement to be based on if params[:categories] is not nil. A cleaner way to do that would be:
```ruby
if params[:categories]
```